File: 1698/1698.go
func numberOfDistinctSubstringsInAS(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := numberOfDistinctSubstringsInAS(root.Left)
	r := numberOfDistinctSubstringsInAS(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 1698/1698.cs
public class Solution {
    public int NumberOfDistinctSubstringsInAS(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(NumberOfDistinctSubstringsInAS(root.left), NumberOfDistinctSubstringsInAS(root.right));
    }
}